chainerx_cc/chainerx/routines/math.cc

```diff
@@ -81,10 +80,25 @@ void BinaryInPlace(Impl&& impl, const Array& x1, Scalar x2) {
     impl(x1, x2, x1);
 }
 
+void CheckArithmeticDtypes(DtypeKind kind1, DtypeKind kind2, DtypeKind out_kind) {
```

Comment:
Write some comment about this function, especially the fact that `Multiply` is an exception.